sup3r/solar/solar.py

```diff
@@ -0,0 +1,630 @@
+# -*- coding: utf-8 -*-
+"""Custom sup3r solar module. This primarily converts GAN output clearsky ratio
+to GHI, DNI, and DHI using NSRDB data and utility modules like DISC
+
+Note that clearsky_ratio is assumed to be clearsky ghi ratio and is calculated
+as daily average GHI / daily average clearsky GHI.
+"""
+import glob
+import json
+import os
+import numpy as np
+import logging
+from scipy.spatial import KDTree
+from farms.disc import disc
+from farms.utilities import calc_dhi, dark_night
+from rex import Resource, MultiTimeResource
+from rex.utilities.fun_utils import get_fun_call_str
+
+from sup3r.utilities import ModuleName
+from sup3r.postprocessing.file_handling import RexOutputs, H5_ATTRS
+
+
+logger = logging.getLogger(__name__)
+
+
+class Solar:
```

Comment:
This is nice work!

Reply:
Thanks! feels so good to have a full day without meetings and a blank slate haha.